### PR TITLE
Migrate to `orchard` crate revision with circuit changes

### PR DESCRIPTION
**Pull request by str4d**
Part of zcash/zcash#5385.

### CODE REVIEW

**Comment by str4d** [comment]
The clippy lint failure is because when we last bumped the Rust version we forgot to bump the linter, so `zcashd` currently uses 1.56.1 (compatible with `incrementalmerkletree` but lints against 1.54.0. I'm fixing this in #5609.

File: .cargo/config.offline [hunk 1]
```diff
@@ -1,9 +1,24 @@
 [source.crates-io]
 replace-with = "vendored-sources"
 
+[source."https://github.com/zcash/incrementalmerkletree.git"]
```
Comment (nuttycom) [inline .cargo/config.offline:4]: I have verified that all these hashes match what I expect.

**Review by nuttycom — APPROVED** [review]
utACK

**Review by ebfull — APPROVED** [review]

**Review by daira — APPROVED** [review]
utACK with comment.

**Pull request merged**